### ISSUE TITLE
Bug - Update Targetdata Scope

### ISSUE THREAD

**Issue by fishme** [aliyun/data-mapping-component#37]
Version: 1.3.9
Scope: update targetData
Expected: add new field in list
Error: TypeError: undefined is not an object (evaluating 'a.id') - Adapter.js sameId line 148. 

I want to add new fields into the target table.
I update the state for the target data. 

```
addField() {
    const target = this.state.targetData.fields;
    target.push({
      name: 'My Name',
      type: 'My Type'
    });

    this.setState({
      targetData: _.cloneDeep(target)
    });
  }
```

```
<DataMapping
              className="container single-no-header"
              sourceClassName="single-no-header-source"
              targetClassName="single-no-header-target"
              columns={this.state.columnData}
              sourceData={this.state.sourceData}
              targetData={this.state.targetData}
              mappingData={this.state.mappingData}
              config={{
                extraPos: {
                  paddingLeft: 10,
                  paddingRight: 10,
                  paddingTop: 10,
                  paddingBottom: 10,
                  paddingCenter: 130
                }
              }}
              onChange={(data) => {
                this.setState({
                  sourceData: _.cloneDeep(data.sourceData),
                  targetData: _.cloneDeep(data.targetData),
                  mappingData: _.cloneDeep(data.mappingData)
                });
              }}
              width={600}
              height={600}
              onRowMouseOver={(row) => {
                //console.log(row);
              }}
              onEdgeClick={(data) => {
                //console.log(data);
              }}
            />
```
extract from my schema.

```
"column":[
        {
           "key":"name",
           "title":"Content",
           "primaryKey":true,
           "width":100,
        },
        {
           "key":"type",
           "title":"Type",
           "width":100
        }
     ],
     "source":{
        "title":"Content",
        "fields":[
           {
              "name":"blueprint",
              "type":"Blueprint"
           }
           ....
          ]},
"target":{
        "title":"Content (Origin)",
        "fields":[
          {
            "name":"Title",
            "type":"String"
         },
....]
```

**Comment by noonnightstorm** [comment]
There are some errors in your code
`
addField() {
    const target =  _.cloneDeep(this.state.targetData);
    target.fields.push({
      name: 'My Name',
      type: 'My Type'
    });

    this.setState({
      targetData: target
    });
  }
`

**Comment by noonnightstorm** [comment]
There are some errors in your code
`

addField() {
const target = _.cloneDeep(this.state.targetData);
target.fields.push({
name: 'My Name',
type: 'My Type'
});

this.setState({
  targetData: target
});
}

`

**Comment by fishme** [comment]
ah, of course! That fixed the error. Thank you. 

It was also not working. 
In your example you use, the field "id" for the identifier. With my case I removed the id and use instead "name". 

In the file: src/adaptor.js -> function diffPropsData
There you use  the "id" property hardcoded for the compare. 

I created a merge request for fix this issue https://github.com/aliyun/data-mapping-component/pull/38 

```
const isSameId = (a, b) => a.id === b.id;
....
 let addResult = _.differenceWith(
        _newNode.fields,
        _.get(_oldNode, 'options.fields'),
        isSameId
      );
```
